### PR TITLE
Added course settings page

### LINKED ISSUE #254: Create the UI for the Course Settings Page
**Goals**

Screen Shot 2022-10-27 at 8 33 57 PM

The final part of finishing the Contributor Experience is allowing contributors to change the privacy of their course, along with deleing the course. For this issue, focus on building out the Settings Page. There will be another issue related to connecting this to our API.

**Steps**

1. Add a dropdown for selecting public/private
2. If private is selected, show an input for the secret password.
3. Add a delete course button.

### PR DESCRIPTION
**Pull request by jamieRollison**
## Status:

:rocket: Ready

## Description

Made the course settings page. Also created a settings button variant in nativebase, which may or may not conflict with the buttons in some other issues

Fixes #254 

## Todos
when I added conditional rendering, the security code moved down the screen a little. It still looks fine, but if we prefer the original design it may be worth the time for someone with more experience to go in and move it up a little.

## Screenshots
826E8494-AD3C-42BF-BA6E-F73C2096B928_1_101_o

### CODE REVIEW

**Review by eeshabarua — CHANGES_REQUESTED** [review]
This looks so good! Except locally I'm getting this weird error but it might j be a my-computer thing
image

File: client/src/pages/CourseSettings/CourseSettings.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import StyledButton from 'components/StyledButton'
+import { Alert, StyleSheet, View } from 'react-native'
+import { MaterialCommunityIcons } from '@expo/vector-icons'
+import i18n from 'utils/i18n'
+
+const styles = StyleSheet.create({
+  description: {
+    marginTop: 10,
+    marginBottom: 20,
+    marginHorizontal: '3%', // i think there's a better way to do this
```
Comment (ashayp22) [inline client/src/pages/CourseSettings/CourseSettings.js:12]: Is this your comment? What exactly are you trying to achieve?

File: client/src/pages/CourseSettings/CourseSettings.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        <Select
+          defaultValue={selected}
+          borderColor="black"
+          height="38%"
+          onValueChange={setSelected}
+        >
+          <Select.Item label={i18n.t('dict.private')} value="private" />
+          <Select.Item label={i18n.t('dict.public')} value="public" />
+        </Select>
+        <Text fontFamily="heading" fontWeight="regular" fontSize="lg">
```
Comment (ashayp22) [inline client/src/pages/CourseSettings/CourseSettings.js:55]: Make sure to only show the input field if the course is private.

File: client/src/pages/ManageVocab/ManageVocab.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ const ManageVocab = ({ navigation }) => {
   const { lessonData, currentCourseId, currentLessonId } = useSelector(
     (state) => state.language,
   )
-  //   const allVocab = lessonData.vocab
+  //   const allVocab = lessonData.vocabs
```
[the remark this answers is not in the document]
Comment (ashayp22) [reply client/src/pages/ManageVocab/ManageVocab.js:18]: +1

File: client/src/pages/ManageVocab/ManageVocab.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ const ManageVocab = ({ navigation }) => {
         _id: item._id,
         title: item.original,
         body: item.translation,
-        indicatorType: INDICATOR_TYPES.NONE,
+        indicatorType: INDICATOR_TYPES.NONE, // TODO: remove hard-coded value
```
Comment (ashayp22) [inline client/src/pages/ManageVocab/ManageVocab.js:35]: Comment should be good to remove since we no longer use indicator types

**Review by ashayp22 — CHANGES_REQUESTED** [review]
Looks great Jamie, I left a few inline comments. Main functional fix is to make sure that the Private Code Input Field is only shown when the Course is set to Private.

**Review by ashayp22 — APPROVED** [review]

**Pull request merged**